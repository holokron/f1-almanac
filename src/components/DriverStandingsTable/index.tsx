import React, { ReactElement } from "react"
import { Table } from "antd"
import {
  Driver,
  DriverStandingsList,
  Team,
  DriverStandingItem,
} from "../../types"

interface DriverStandingsTableProps {
  data: DriverStandingsList
}

function DriverStandingsTable({
  data,
}: DriverStandingsTableProps): ReactElement<DriverStandingsTableProps> {
  return (
    <Table
      bodyStyle={{
        overflowX: "scroll",
      }}
      showHeader
      pagination={false}
      size="small"
      rowKey={(item: DriverStandingItem) => item.Driver.driverId}
      columns={[
        {
          title: "Position",
          dataIndex: "position",
        },
        {
          title: "Driver",
          dataIndex: "Driver",
          render: (driver: Driver) =>
            `${driver.givenName} ${driver.familyName}`,
        },
        {
          title: "Nationality",
          dataIndex: "Driver",
          render: (driver: Driver) => driver.nationality,
          key: "Nationality",
        },
        {
          title: "Team",
          dataIndex: "Constructors",
          render: (teams: Team[]) =>
            teams.map((team: Team) => team.name).join(", "),
        },
        {
          title: "Wins",
          dataIndex: "wins",
        },
        {
          title: "Points",
          dataIndex: "points",
        },
      ]}
      dataSource={data}
    />
  )
}

export default React.memo(DriverStandingsTable)
