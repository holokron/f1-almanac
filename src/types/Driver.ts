export interface DriverListItem {
  driverId: string
  permanentNumber?: string
  code?: string
  url: string
  givenName: string
  familyName: string
  dateOfBirth: string
  nationality: string
}

export type DriversList = DriverListItem[]
