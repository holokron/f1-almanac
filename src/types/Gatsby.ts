export interface SiteMetadata {
  author: string
  copyright: string
  description: string
  keywords: string
  title: string
  version: string
}
