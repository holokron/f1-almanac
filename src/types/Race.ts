import { Circuit } from "./Circuit"

export interface Race {
  season: string
  round: string
  url: string
  raceName: string
  Circuit: Circuit
  date: string
  time: string
}

export type RacesList = Race[]
