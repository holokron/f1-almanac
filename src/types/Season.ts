export interface SeasonListItem {
  season: string
  url: string
}

export type SeasonsList = SeasonListItem[]
