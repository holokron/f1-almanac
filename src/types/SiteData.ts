export interface SiteDataItem {
  link: string
  title: string
}
