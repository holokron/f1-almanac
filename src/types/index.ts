export * from "./Breadcrumb"
export * from "./Circuit"
export * from "./Driver"
export * from "./Location"
export * from "./Race"
export * from "./Season"
export * from "./SiteData"
export * from "./Standings"
export * from "./Team"
