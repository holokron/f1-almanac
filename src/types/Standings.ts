import { Driver } from "./Driver"
import { TeamsList, Team } from "./Team"

export interface DriverStandingItem {
  position: string
  positionText: string
  points: string
  wins: string
  Driver: Driver
  Constructors: TeamsList
}

export type DriverStandingsList = DriverStandingItem[]

export interface TeamStandingItem {
  position: string
  positionText: string
  points: string
  wins: string
  Constructor: Team
}

export type TeamStandingsList = TeamStandingItem[]
