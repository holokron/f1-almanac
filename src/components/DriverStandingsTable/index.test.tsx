import React from "react"
import renderer from "react-test-renderer"
import DriverStandingsTable from "."
import { DriverStandingsList } from "../../types"

describe("components/DriverStandingsTable", (): void => {
  it.each([
    [[]],
    [
      [
        {
          position: "1",
          positionText: "1",
          points: "123",
          wins: "321",
          Constructors: [
            {
              constructorId: "foo",
              name: "Foo",
              nationality: "bar",
              url: "https://foo.bar",
            },
          ],
          Driver: {
            driverId: "foo",
            permanentNumber: "13",
            code: "FOO",
            url: "https://foo.bar",
            givenName: "Foo",
            familyName: "Bar",
            dateOfBirth: "1980-01-01",
            nationality: "bar",
          },
        },
      ],
    ],
  ])(
    "data: %p",
    (data: DriverStandingsList): void => {
      const tree = renderer.create(<DriverStandingsTable data={data} />)

      expect(tree.toJSON()).toMatchSnapshot()
    },
  )
})
