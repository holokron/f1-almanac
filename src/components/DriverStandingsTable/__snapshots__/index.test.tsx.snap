// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/DriverStandingsTable data: [[Object]] 1`] = `
<div
  className="ant-table-wrapper"
>
  <div
    className="ant-spin-nested-loading"
  >
    <div
      className="ant-spin-container"
    >
      <div
        className="ant-table ant-table-small ant-table-scroll-position-left"
        style={Object {}}
      >
        <div
          className="ant-table-content"
        >
          <div
            className="ant-table-body"
            onScroll={[Function]}
            onWheel={[Function]}
            style={
              Object {
                "overflowX": "scroll",
              }
            }
          >
            <table
              className=""
              style={Object {}}
            >
              <colgroup>
                <col
                  style={
                    Object {
                      "minWidth": undefined,
                      "width": undefined,
                    }
                  }
                />
                <col
                  style={
                    Object {
                      "minWidth": undefined,
                      "width": undefined,
                    }
                  }
                />
                <col
                  style={
                    Object {
                      "minWidth": undefined,
                      "width": undefined,
                    }
                  }
                />
                <col
                  style={
                    Object {
                      "minWidth": undefined,
                      "width": undefined,
                    }
                  }
                />
                <col
                  style={
                    Object {
                      "minWidth": undefined,
                      "width": undefined,
                    }
                  }
                />
                <col
                  style={
                    Object {
                      "minWidth": undefined,
                      "width": undefined,
                    }
                  }
                />
              </colgroup>
              <thead
                className="ant-table-thead"
              >
                <tr
                  style={
                    Object {
                      "height": null,
                    }
                  }
                >
                  <th
                    className=""
                  >
                    <span
                      className="ant-table-header-column"
                    >
                      <div>
                        <span
                          className="ant-table-column-title"
                        >
                          Position
                        </span>
                        <span
                          className="ant-table-column-sorter"
                        />
                      </div>
                    </span>
                  </th>
                  <th
                    className=""
                  >
                    <span
                      className="ant-table-header-column"
                    >
                      <div>
                        <span
                          className="ant-table-column-title"
                        >
                          Driver
                        </span>
                        <span
                          className="ant-table-column-sorter"
                        />
                      </div>
                    </span>
                  </th>
                  <th
                    className=""
                  >
                    <span
                      className="ant-table-header-column"
                    >
                      <div>
                        <span
                          className="ant-table-column-title"
                        >
                          Nationality
                        </span>
                        <span
                          className="ant-table-column-sorter"
                        />
                      </div>
                    </span>
                  </th>
                  <th
                    className=""
                  >
                    <span
                      className="ant-table-header-column"
                    >
                      <div>
                        <span
                          className="ant-table-column-title"
                        >
                          Team
                        </span>
                        <span
                          className="ant-table-column-sorter"
                        />
                      </div>
                    </span>
                  </th>
                  <th
                    className=""
                  >
                    <span
                      className="ant-table-header-column"
                    >
                      <div>
                        <span
                          className="ant-table-column-title"
                        >
                          Wins
                        </span>
                        <span
                          className="ant-table-column-sorter"
                        />
                      </div>
                    </span>
                  </th>
                  <th
                    className=""
                  >
                    <span
                      className="ant-table-header-column"
                    >
                      <div>
                        <span
                          className="ant-table-column-title"
                        >
                          Points
                        </span>
                        <span
                          className="ant-table-column-sorter"
                        />
                      </div>
                    </span>
                  </th>
                </tr>
              </thead>
              <tbody
                className="ant-table-tbody"
              >
                <tr
                  className="ant-table-row ant-table-row-level-0"
                  data-row-key="foo"
                  onClick={[Function]}
                  onContextMenu={[Function]}
                  onDoubleClick={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  style={
                    Object {
                      "height": null,
                    }
                  }
                >
                  <td
                    className=""
                    onClick={[Function]}
                  >
                    <span
                      className="ant-table-row-indent indent-level-0"
                      style={
                        Object {
                          "paddingLeft": "0px",
                        }
                      }
                    />
                    1
                  </td>
                  <td
                    className=""
                    onClick={[Function]}
                  >
                    Foo Bar
                  </td>
                  <td
                    className=""
                    onClick={[Function]}
                  >
                    bar
                  </td>
                  <td
                    className=""
                    onClick={[Function]}
                  >
                    Foo
                  </td>
                  <td
                    className=""
                    onClick={[Function]}
                  >
                    321
                  </td>
                  <td
                    className=""
                    onClick={[Function]}
                  >
                    123
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`components/DriverStandingsTable data: [] 1`] = `
<div
  className="ant-table-wrapper"
>
  <div
    className="ant-spin-nested-loading"
  >
    <div
      className="ant-spin-container"
    >
      <div
        className="ant-table ant-table-small ant-table-empty ant-table-scroll-position-left"
        style={Object {}}
      >
        <div
          className="ant-table-content"
        >
          <div
            className="ant-table-body"
            onScroll={[Function]}
            onWheel={[Function]}
            style={
              Object {
                "overflowX": "scroll",
              }
            }
          >
            <table
              className=""
              style={Object {}}
            >
              <colgroup>
                <col
                  style={
                    Object {
                      "minWidth": undefined,
                      "width": undefined,
                    }
                  }
                />
                <col
                  style={
                    Object {
                      "minWidth": undefined,
                      "width": undefined,
                    }
                  }
                />
                <col
                  style={
                    Object {
                      "minWidth": undefined,
                      "width": undefined,
                    }
                  }
                />
                <col
                  style={
                    Object {
                      "minWidth": undefined,
                      "width": undefined,
                    }
                  }
                />
                <col
                  style={
                    Object {
                      "minWidth": undefined,
                      "width": undefined,
                    }
                  }
                />
                <col
                  style={
                    Object {
                      "minWidth": undefined,
                      "width": undefined,
                    }
                  }
                />
              </colgroup>
              <thead
                className="ant-table-thead"
              >
                <tr
                  style={
                    Object {
                      "height": null,
                    }
                  }
                >
                  <th
                    className=""
                  >
                    <span
                      className="ant-table-header-column"
                    >
                      <div>
                        <span
                          className="ant-table-column-title"
                        >
                          Position
                        </span>
                        <span
                          className="ant-table-column-sorter"
                        />
                      </div>
                    </span>
                  </th>
                  <th
                    className=""
                  >
                    <span
                      className="ant-table-header-column"
                    >
                      <div>
                        <span
                          className="ant-table-column-title"
                        >
                          Driver
                        </span>
                        <span
                          className="ant-table-column-sorter"
                        />
                      </div>
                    </span>
                  </th>
                  <th
                    className=""
                  >
                    <span
                      className="ant-table-header-column"
                    >
                      <div>
                        <span
                          className="ant-table-column-title"
                        >
                          Nationality
                        </span>
                        <span
                          className="ant-table-column-sorter"
                        />
                      </div>
                    </span>
                  </th>
                  <th
                    className=""
                  >
                    <span
                      className="ant-table-header-column"
                    >
                      <div>
                        <span
                          className="ant-table-column-title"
                        >
                          Team
                        </span>
                        <span
                          className="ant-table-column-sorter"
                        />
                      </div>
                    </span>
                  </th>
                  <th
                    className=""
                  >
                    <span
                      className="ant-table-header-column"
                    >
                      <div>
                        <span
                          className="ant-table-column-title"
                        >
                          Wins
                        </span>
                        <span
                          className="ant-table-column-sorter"
                        />
                      </div>
                    </span>
                  </th>
                  <th
                    className=""
                  >
                    <span
                      className="ant-table-header-column"
                    >
                      <div>
                        <span
                          className="ant-table-column-title"
                        >
                          Points
                        </span>
                        <span
                          className="ant-table-column-sorter"
                        />
                      </div>
                    </span>
                  </th>
                </tr>
              </thead>
              <tbody
                className="ant-table-tbody"
              />
            </table>
          </div>
          <div
            className="ant-table-placeholder"
          >
            <div
              className="ant-empty ant-empty-normal"
            >
              <div
                className="ant-empty-image"
              >
                <img
                  alt="No Data"
                  src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNjQiIGhlaWdodD0iNDEiIHZpZXdCb3g9IjAgMCA2NCA0MSIgIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyI+CiAgPGcgdHJhbnNmb3JtPSJ0cmFuc2xhdGUoMCAxKSIgZmlsbD0ibm9uZSIgZmlsbC1ydWxlPSJldmVub2RkIj4KICAgIDxlbGxpcHNlIGZpbGw9IiNGNUY1RjUiIGN4PSIzMiIgY3k9IjMzIiByeD0iMzIiIHJ5PSI3Ii8+CiAgICA8ZyBmaWxsLXJ1bGU9Im5vbnplcm8iIHN0cm9rZT0iI0Q5RDlEOSI+CiAgICAgIDxwYXRoIGQ9Ik01NSAxMi43Nkw0NC44NTQgMS4yNThDNDQuMzY3LjQ3NCA0My42NTYgMCA0Mi45MDcgMEgyMS4wOTNjLS43NDkgMC0xLjQ2LjQ3NC0xLjk0NyAxLjI1N0w5IDEyLjc2MVYyMmg0NnYtOS4yNHoiLz4KICAgICAgPHBhdGggZD0iTTQxLjYxMyAxNS45MzFjMC0xLjYwNS45OTQtMi45MyAyLjIyNy0yLjkzMUg1NXYxOC4xMzdDNTUgMzMuMjYgNTMuNjggMzUgNTIuMDUgMzVoLTQwLjFDMTAuMzIgMzUgOSAzMy4yNTkgOSAzMS4xMzdWMTNoMTEuMTZjMS4yMzMgMCAyLjIyNyAxLjMyMyAyLjIyNyAyLjkyOHYuMDIyYzAgMS42MDUgMS4wMDUgMi45MDEgMi4yMzcgMi45MDFoMTQuNzUyYzEuMjMyIDAgMi4yMzctMS4zMDggMi4yMzctMi45MTN2LS4wMDd6IiBmaWxsPSIjRkFGQUZBIi8+CiAgICA8L2c+CiAgPC9nPgo8L3N2Zz4K"
                />
              </div>
              <p
                className="ant-empty-description"
              >
                No Data
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
